doc/workflow.md

```diff
@@ -0,0 +1,66 @@
+# Workflow
```

Comment:
This is very "developer of kube-spawn"-focused, detailing each steps how it is implemented instead of how it can be used. I think we need a workflow documentation for the point of view of users too.

I created the directory [doc/devel](https://github.com/kinvolk/kube-spawn/tree/master/doc/devel) for developer-focused documentation and `doc` should be for users in my opinion.